### PR TITLE
pdf-parser: init at 0.7.4

### PR DESCRIPTION
**Pull request by lightdiscord**
###### Motivation for this change

Add missing package for #81418 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/pdf-parser/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+  format = "other";
+
+  installPhase = ''
+    install -Dm555 pdf-parser.py $out/bin/pdf-parser.py
+  '';
+
+  preFixup = ''
+    substituteInPlace $out/bin/pdf-parser.py \
+      --replace '/usr/bin/python' '${python3Packages.python}/bin/python'
```
Comment (jtojnar) [inline pkgs/tools/misc/pdf-parser/default.nix]: Is this necessary? I would expect `buildPythonApplication` to fix this automatically.

Comment (lightdiscord) [reply pkgs/tools/misc/pdf-parser/default.nix]: The `buildPythonApplication` only fixes the shebang at the first line of the script.
But without this line, there is still a `/usr/bin/python` in the script.

Comment (Ma27) [reply pkgs/tools/misc/pdf-parser/default.nix]: IMHO `buildPythonApplication` shouldn't touch the code itself and it's perfectly fine to manually fix those paths in the sources, so LGTM.

**Review by Ma27 — APPROVED** [review]

**Comment by FRidh** [comment]
please rebase

**Comment by lightdiscord** [comment]
Rebase done. I don't know why but I think ofborg is stuck. :thinking:

**Pull request merged**